### ISSUE TITLE
Memory leak ?

### ISSUE THREAD

**Issue by NotSqrt** [ijl/orjson#4]
Hello !

I've just tested on data that look like this :

```python
dumped = '{"12999":-634,"13000":-84,"13001":-789,"13002":164,"13003":671,"13004":4262,"13005":4376,"13006":481,"13007":5469,"13008":613,"13009":4736,"13010":4218,"13011":517,"13012":5473,"13013":5404,"13014":4157,"13015":3919,"13016":-497,"13017":580,"13018":3298,"13019":-1018,"13020":-838,"13021":2900,"13022":2862,"13023":828,"13024":3526,"13025":3349,"13026":1051,"13027":1463,"13028":1978,"13029":1316,"13030":4445,"13031":4316,"13032":34,"13033":1861,"13034":5046,"13035":1370,"13036":798,"13037":-679,"13038":3700,"13039":1878,"13040":5263,"13041":5396,"13042":4714,"13043":2642,"13044":5224,"13045":416,"13046":4643,"13047":4461,"13048":5611,"13049":2421,"13050":-68,"13051":2717,"13052":2826,"13053":3578,"13054":-707,"13055":4012,"13056":319,"13057":703,"13058":-1691,"13059":-927,"13060":5496,"13061":4991,"13062":-1246,"13063":320,"13064":2282,"13065":5002,"13066":1944,"13067":3033,"13068":-284,"13069":217,"13070":2993,"13071":2696,"13072":401,"13073":2467,"13074":2410,"13075":-679,"13076":210,"13077":3479,"13078":-1001,"13079":-745,"13080":90,"13081":-1087,"13082":2623,"13083":5934,"13084":-1202,"13085":4531,"13086":3787,"13087":-361,"13088":1814,"13089":584,"13090":-443,"13091":4458,"13092":3784,"13093":-380,"13094":2258,"13095":1905,"13096":1605,"13097":41,"13098":5859}'
%timeit orjson.loads(dumped)
```

But I see that memory usage increases a lot for the python process, and does not go down.

Thanks !

**Comment by NotSqrt** [comment]
This is with python 3.6.8, `orjson-1.2.0-cp36-cp36m-manylinux1_x86_64.whl`.

**Comment by NotSqrt** [comment]
The %timeit was just a fast way to write a loop.
There is a memory leak, I crashed my computer when benchmarking orjson vs ujson (with python 3.6 and python 3.7):

```python
import gc, ujson, orjson, psutil
proc = psutil.Process()
print(f'{proc.memory_percent():.3f}%')

dumped = '{"12999":-634,"13000":-84,"13001":-789,"13002":164,"13003":671,"13004":4262,"13005":4376,"13006":481,"13007":5469,"13008":613,"13009":4736,"13010":4218,"13011":517,"13012":5473,"13013":5404,"13014":4157,"13015":3919,"13016":-497,"13017":580,"13018":3298,"13019":-1018,"13020":-838,"13021":2900,"13022":2862,"13023":828,"13024":3526,"13025":3349,"13026":1051,"13027":1463,"13028":1978,"13029":1316,"13030":4445,"13031":4316,"13032":34,"13033":1861,"13034":5046,"13035":1370,"13036":798,"13037":-679,"13038":3700,"13039":1878,"13040":5263,"13041":5396,"13042":4714,"13043":2642,"13044":5224,"13045":416,"13046":4643,"13047":4461,"13048":5611,"13049":2421,"13050":-68,"13051":2717,"13052":2826,"13053":3578,"13054":-707,"13055":4012,"13056":319,"13057":703,"13058":-1691,"13059":-927,"13060":5496,"13061":4991,"13062":-1246,"13063":320,"13064":2282,"13065":5002,"13066":1944,"13067":3033,"13068":-284,"13069":217,"13070":2993,"13071":2696,"13072":401,"13073":2467,"13074":2410,"13075":-679,"13076":210,"13077":3479,"13078":-1001,"13079":-745,"13080":90,"13081":-1087,"13082":2623,"13083":5934,"13084":-1202,"13085":4531,"13086":3787,"13087":-361,"13088":1814,"13089":584,"13090":-443,"13091":4458,"13092":3784,"13093":-380,"13094":2258,"13095":1905,"13096":1605,"13097":41,"13098":5859}'

for __ in range(100_000):
    orjson.loads(dumped)
gc.collect()

print(f'{proc.memory_percent():.3f}%')

for __ in range(100_000):
    ujson.loads(dumped)

gc.collect()

print(f'{proc.memory_percent():.3f}%')

```

gives :
```
0.076%  # initial RAM usage 
5.316%  # after orjson
5.316%  # after ujson + orjson
```

This memory is not freed.
Thanks.

**Comment by ijl** [comment]
Ok, thanks for the reproduction. I'll look into it.

**Comment by ijl** [comment]
Fixed in fb5bdbea78b5d0801f9bb6fd073b4171036026ca, released in 1.2.1.
